add titiler endpoint from cdk-pgstac construct

emileten opened the pull request:
Adds a `titiler-pgstac` construct for mosaics built from and saved directly to the pgstac database, with a lambda exposed on an HTTP API. I simply import the construct developed here https://github.com/developmentseed/cdk-pgstac/pull/42. I deployed to `test` and the tiler API health checks pass (they check the connection to the pgstac database). 

Since the latter is not yet released, for now I deployed with a local build of cdk-pgstac (the setup for the local build is not showing up in the diff -- I will simply bump the cdk-pgstac version in the `package.json` file here once the release is made). The reason I did not wait for the release is that it might take some time for people to review my PR. 

Things to note : 

- [x] the lambda is deployed in the same VPC and subnet as the database
- [ ] I modified the lambda role policies to grant it access to `nasa-maap-data-store` (see the `buckets` parameter in the diff). Is there any other bucket we need this lambda to have access to ?

jjfrench commented on cdk/PgStacInfra.ts:
```diff
@@ -57,6 +58,21 @@ export class PgStacInfra extends Stack {
     });
 
 
+    const buckets = ["nasa-maap-data-store"];
```
Should this be from the configuration file?

emileten replied:
Yep that would be better.

jjfrench replied:
I think that my only comment is still just this configuration should be moved

emileten replied:
Thanks @jjfrench.

Just pushed a commit that addresses this.

emileten commented:
@jjfrench anything else ? Addressed your comment. 

Once someone approves this and #6 is merged to `main`, I will pull from `main` and deploy this branch.

jjfrench commented:
I approved #6, I'll review recent changes on this one now 🎂

emileten commented:
Note that I haven't tested the latest change, there may be bugs -- I don't know typescript.

jjfrench reviewed (approved):
I don't know typescript either but the changes in theory look good to me ¯\_(ツ)_/¯

emileten commented:
So many github actions running to solve bugs... Should learn how to use https://github.com/nektos/act..

Merging this !

jjfrench commented on .env:
```diff
@@ -1,2 +1,3 @@
\ No newline at end of file
+STAGE='test'
+DB_ALLOCATED_STORAGE=20
+STAC_API_INTEGRATION_API_ARN='arn:aws:execute-api:us-west-2:532321095167:hqprpfkrqg/*/$default'
```
I think this was added after approval, but can this be moved to github env? The public can still see .env files

emileten replied:
@jjfrench this is not used, we're solely relying on github env/secrets https://github.com/MAAP-Project/maap-eoapi/blob/007d514cc86a5f4dca72d76e3c0fea1aac097818/.github/workflows/deploy.yml#L47. 

I am going to delete this, along with `deploy.sh`.
